feat(connectors, soap): added SOAP connector doc (#3822)
camunda/camunda-docs: docs/components/connectors/protocol/soap.md

---
id: soap
title: SOAP Connector
sidebar_label: SOAP Connector
description: The SOAP Connector allows you to connect your BPMN with SOAP services.
---

**SOAP** is a messaging protocol specification for exchanging structured
information in the implementation of web services in computer networks.

The **SOAP Connector** allows you to interact with [SOAP](https://en.wikipedia.org/wiki/SOAP) services endpoints
from your BPMN process.

## Prerequisites

The **SOAP Connector** is only supported by self-managed Camunda instances.

To use the **SOAP Connector**, you need to have an active SOAP service.

## Create a SOAP Connector task

> To use the **SOAP Connector** in your process, either change the type of existing task by clicking on it and using

Author: ```suggestion
To use the **SOAP Connector** in your process, either change the type of an existing task by clicking on it and using
```

From the **SOAP body** dropdown, select whether you will provide the SOAP request body in a form of **Template**, or
**XML compatible JSON**.

#### Template

When **Template** is chosen, you have to enter the **XML template** value, e.g.
`<camunda:Param><camunda:ParamType>{{paramValue}}</camunda:ParamType></camunda:Param>`.

You have to enter the **XML template context** value, e.g.
`={paramValue: 1234567890}`.

You have to enter the **Namespaces** value, e.g.
`={"camunda":"http://my.service.com/webservicesserver/"}`.

#### XML compatible JSON

When **XML compatible JSON** is chosen, you have to enter the **JSON definition**, e.g.

```xml
<camunda:Object01>
  <camunda:Object02>{{myObjectValue}}</camunda:Object02>
</camunda:Object01>
```

You have to enter the **Namespaces** value, e.g.
`={"camunda":"http://my.service.com/webservicesserver/"}`.

## Output mapping

### Result variable

You can export a complete response from an SOAP call into a dedicated variable accessible anywhere in a process.
To do so, just input a variable name in the **Result variable** field. We recommend using a unique name to avoid
variables being overwritten.

## Usage examples

### Example 1

> For example, you would like to send a following SOAP request:

Reviewer: "a following SOAP request" should be "the following SOAP request"

URL: `https://myservice:8888/webservice.wso`

Body:

```xml
<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
  <soap:Body>
    <Object01>
      <Object02>12345</Object02>
    </Object01>
  </soap:Body>
</soap:Envelope>
```

In order to do so, in your BPMN diagram, set the field **Service URL** as `https://myservice:8888/webservice.wso`, and **SOAP body** as

```json
{
  "Object01": {
    "Object02": 12345
  }
}
```

### Example 2: pre-defined namespaces

Consider a namespace is defined within your objects, and you wish to send the following request:

URL: `https://myservice:8888/webservice.wso`

Body:

```xml
<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
  <soap:Body>

>     <Object01 mlns="http://www.my.namespace.com/namespace/">

Reviewer: Should this be xmlns or mlns?

      <Object02>12345</Object02>
    </Object01>
  </soap:Body>
</soap:Envelope>
```

In order to do so, in your BPMN diagram, set the field **Service URL** as `https://myservice:8888/webservice.wso`, and **SOAP body** as

```json
{
  "ns:Object01": {
    "ns:Object02": 12345
  }
}
```

> Please, pay attention, that here we introduced a new `ns:` prefix. Prefix, can be any arbitrary string, that is not defined as namespace.

Reviewer: Use softer language, something like "Please note that here we introduced a new ns: prefix."

Now, you'll need to associate a namespace. You can do it by setting the following value at the **Namespaces** field.
For the given example, it should be set as:

```json
{
  "ns": "http://www.my.namespace.com/namespace/"
}
```

### Example 3: using templates

As an alternative, you can use templates to send SOAP messages.

URL: `https://myservice:8888/webservice.wso`

Body:

```xml
<?xml version="1.0" encoding="utf-8"?>
<soap:Envelope xmlns:soap="http://schemas.xmlsoap.org/soap/envelope/">
  <soap:Body>
    <Object01>
      <Object02>12345</Object02>
    </Object01>
  </soap:Body>
</soap:Envelope>
```

> For that, set the **SOAP body** dropdown to **Template**.

Reviewer: "For that, set the SOAP body dropdown to Template." could be "To do this, set the SOAP body dropdown to Template."

---
id: soap
title: SOAP Connector
sidebar_label: SOAP Connector
description: The SOAP Connector allows you to connect your BPMN process with SOAP services.
---

:::note

> The **SOAP Connector** is only supported by Self-Managed Camunda 8 instances.

Reviewer 2: It is also supported in hybrid mode

Author: Updated

:::

**Simple Object Access Protocol (SOAP)** is a messaging protocol specification for exchanging structured
information in the implementation of web services in computer networks.

> The **SOAP Connector** allows you to interact with [SOAP](https://en.wikipedia.org/wiki/SOAP) service endpoints

Reviewer 3: Is this the best link we have for info on SOAP? IBM and Oracle have some docs on this too, but Wikipedia might be the best we can do for now.

Author: AFAIK, IBM and Oracle have their sort of "proprietary" flavors. If wikipedia is not reliable source here, maybe consider w3c https://www.w3.org/TR/soap/?

Reviewer 4: I'm good with either external link. Since SOAP is a protocol, everyone wants their own flavor 🍦

Author: I added the `https://www.w3.org/TR/soap/` link. I believe this one is the most generic one

Follow our [guide to using Connectors](/components/connectors/use-connectors/index.md) to learn more.

## Connection

Enter your SOAP service URL in the field **Service URL**, for example `https://myservice.com/service/MyService.wso`.

## Authentication

Select the authentication type from the **Authentication** dropdown.

### None

Use **None** if the SOAP service does not require authentication.

### WSS username token

Use **WSS username token** in the **Authentication** dropdown when the requested SOAP endpoint requires
[username token extension](https://docs.oasis-open.org/wss/v1.1/wss-v1.1-spec-pr-UsernameTokenProfile-01.htm#_Toc104276211).

Enter **Username**, **Password**, and indicate if the password is encoded.

:::note
The **SOAP Connector** currently supports only `SHA-1` password encoding.
:::

### WSS signature

Use the **WSS signature** in the **Authentication** dropdown when the requested SOAP endpoint requires a message to be
cryptographically signed with a [signature](http://docs.oasis-open.org/wss-m/wss/v1.1.1/cs01/wss-SOAPMessageSecurity-v1.1.1-cs01.html#_Toc307407954).

> Enter all necessary fields according to your service specification.

Reviewer 2: We should be prepared to extend here based on customer feedback :P

Author: I don't have a valid example. Do you mind adding one?

Reviewer 4: @jonathanlukas, do you have something already? Or can we resolve this?

Reviewer 2: not yet

## SOAP message

### SOAP version

> Select the latest supported version of the SOAP service.

Reviewer 2: ```suggestion
Select the desired version of the SOAP service.
```

Author: Updated

### SOAPAction HTTP header

Enter the SOAPAction HTTP header that will be used in the request. Leave this value blank if the SOAPAction HTTP header

> won't be used in your request.

Reviewer 2: This is only required in SOAP 1.1

Author: Updated

### SOAP header

From the dropdown, select whether the **SOAP header** is required, and if so, in which format you wish to provide it.

### SOAP body

From the **SOAP body** dropdown, select whether you will provide the SOAP request body in a form of **Template**, or
**XML compatible JSON**.

#### Template

When **Template** is chosen, enter the **XML template** value, for example `<camunda:Param><camunda:ParamType>{{paramValue}}</camunda:ParamType></camunda:Param>`.

Enter the **XML template context** value, for example `={paramValue: 1234567890}`, and enter the **Namespaces** value, for example `={"camunda":"http://my.service.com/webservicesserver/"}`.

#### XML compatible JSON

When **XML compatible JSON** is chosen, enter the **JSON definition**, for example

> ```xml

Reviewer 2: This is XML, I would have expected a json document here, plus its "transformed" xml equivalent.

<camunda:Object01>
  <camunda:Object02>{{myObjectValue}}</camunda:Object02>
</camunda:Object01>
```

Enter the **Namespaces** value, for example `={"camunda":"http://my.service.com/webservicesserver/"}`.

## Output mapping

### Result variable

You can export a complete response from a SOAP call into a dedicated variable accessible anywhere in a process.
To do so, input a variable name in the **Result variable** field. Use a unique name to avoid

> overwriting variables.

Reviewer 2: Here should add the schema of the output here:
```json
{
  "response": {
    "Envelope": {
      "Header": {
        <header content>
      },
      "Body": {
        <body content>
      } 
    }
  }
}
```

Author: Updated this section.

> ## Usage examples

Reviewer 2: Can we have an example with response mapping as well or is this unusual?

Author: Added this section

---
id: soap
title: SOAP Connector
sidebar_label: SOAP Connector
description: The SOAP Connector allows you to connect your BPMN process with SOAP services.
---

:::note

> The **SOAP Connector** is only supported by Self-Managed and Hybrid Camunda 8 instances.

Reviewer 3: What do we mean when we say "Hybrid Camunda 8"? Is this the same as Self-Managed?

Reviewer 2: Hybrid means that you are running the cluster in Saas, but your connector runtime is local.

Reviewer 3: @akeller Do we have this defined anywhere, and I'm just missing it? If not, can add to glossary if this is a term we may use more often.

Author: @christinaausley we added it here previously: https://docs.camunda.io/docs/guides/use-connectors-in-hybrid-mode/

Author: I mean, introduced the term

Reviewer 3: Ah! Got it. I may open an issue in the backlog to link to this from within the Connector docs and define this in our glossary. My mind went to badges, and wondering how we would flag things moving forward -- in this case, we can just lowercase hybrid and link to the hybrid mode guide when referenced in the Connector docs. WDYT?

Author: Works for me

Reviewer 4: To my knowledge, Connectors is the only feature used in hybrid mode, but it would be great to have it in the glossary @christinaausley!

## SOAP message

### SOAP version

Select the desired version of the SOAP service.

### SOAPAction HTTP header

Enter the SOAPAction HTTP header that will be used in the request. Leave this value blank if the SOAPAction HTTP header
won't be used in your request. This field is only required by SOAP version 1.1.

### SOAP header

From the dropdown, select whether the **SOAP header** is required, and if so, in which format you wish to provide it.

### SOAP body

From the **SOAP body** dropdown, select whether you will provide the SOAP request body in a form of **Template**, or
**XML compatible JSON**.

#### Template

When **Template** is chosen, enter the **XML template** value, for example `<camunda:Param><camunda:ParamType>{{paramValue}}</camunda:ParamType></camunda:Param>`.

Enter the **XML template context** value, for example `={paramValue: 1234567890}`, and enter the **Namespaces** value, for example `={"camunda":"http://my.service.com/webservicesserver/"}`.

#### XML compatible JSON

When **XML compatible JSON** is chosen, enter the **JSON definition**, for example

```json
= {
  "camunda:Object01": {
    "camunda:Object02": myObjectValue
  }
}
```

> Enter the **Namespaces** value, for example `={"camunda":"http://my.service.com/webservicesserver/"}`.

Reviewer 2: ```suggestion
#### Namespaces

All namespaces required to create a valid SOAP request should be added here.

Enter the **Namespaces** value, for example `={"camunda":"http://my.service.com/webservicesserver/"}`.
```
